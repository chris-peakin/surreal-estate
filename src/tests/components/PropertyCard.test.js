import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCardComponent";

describe("PropertyCard", () => {
  const houseInfo = {
    title: "2bedhouse",
    type: "Detached",
    bedrooms: 2,
    bathrooms: 1,
    price: 100000,
    city: "Manchester",
    email: "joebloggs@gmail.com",
  };

  const { asFragment, getByText, getByTestId } = render(
    <PropertyCard houseInfo={houseInfo} />
  );

  test("renders correctly", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  test("provided prop has correct values", () => {
    expect(houseInfo.title).toBe("2bedhouse");
    expect(houseInfo.type).toBe("Detached");
    expect(houseInfo.bedrooms).toBe(2);
    expect(houseInfo.bathrooms).toBe(1);
    expect(houseInfo.price).toBe(100000);
    expect(houseInfo.city).toBe("Manchester");
    expect(houseInfo.email).toBe("joebloggs@gmail.com");
  });

  test("renders correct value for props", () => {
    expect(getByText(houseInfo.title)).toHaveClass("property-card__title");
  });
});
