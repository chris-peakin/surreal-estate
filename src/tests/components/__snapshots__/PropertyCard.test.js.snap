// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
    data-testid="property-card"
  >
    <div
      class="title"
      data-testid="property-card__title"
    />
    <div
      class="type-city"
      data-testid="property-card__type"
    />
    <div
      class="type-city"
      data-testid="property-card__city"
    />
    <div
      class="bathrooms"
      data-testid="property-card__bathrooms"
    >
      Bathrooms: 
    </div>
    <div
      class="bedrooms"
      data-testid="property-card__bedrooms"
    >
      Bedrooms: 
    </div>
    <div
      class="price"
      data-testid="property-card__price"
    >
      Â£
    </div>
    <div
      class="email"
      data-testid="property-card__email"
    >
      <a
        href="mailto:undefined"
      >
        Send email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
